monkey/tests/unit_tests/monkey_island/cc/services/authentication_service/test_authentication_service.py

```diff
@@ -116,6 +116,47 @@ def test_handle_sucessful_login(
     assert mock_repository_encryptor.unlock.call_args[0][0] != PASSWORD
 
 
+def test_generate_new_token_pair__generates_tokens(
+    mock_token_generator: TokenGenerator,
+    mock_token_validator: TokenValidator,
+    authentication_facade: AuthenticationFacade,
+):
+    refresh_token = "original_token"
+    user = User(username=USERNAME, password=PASSWORD, fs_uniquifier="a")
+    user.save()
+    mock_token_generator.generate_token.return_value = "new_token"
+    mock_token_validator.get_token_payload.return_value = "a"
+    access_token, new_refresh_token = authentication_facade.generate_new_token_pair(refresh_token)
+
+    assert new_refresh_token != refresh_token
+    assert access_token != refresh_token
+    assert access_token != new_refresh_token
```

Comment:
Missing a check or test that verifies the newly returned authentication token is different from the user's original authentication token.